stglib/vec/dat2cdf.py

```diff
@@ -206,8 +225,29 @@ def load_dat(basefile):
         "AnalogInput2",
         "Checksum",
     ]
+
     dat = pd.read_csv(f"{basefile}.dat", header=None, sep=r"\s+", names=names)
-    dat = dat.set_index(["Burst", "Ensemble"])
+
+    if metadata["instmeta"]["VECBurstInterval"] != "CONTINUOUS":
+        array = np.zeros(np.shape(dat.VEL1))
+
+        for i in range(np.shape(dsvhd["time"])[0]):
+            index = np.where(dat.Burst == i + 1)
+            array[index] = dsvhd["time"][i]
+
+        array = pd.to_datetime(array)
+        frequency = 1000 / metadata["instmeta"]["VECSamplingRate"]
+        delta = frequency * (dat.Ensemble.values - 1)
+        dat["time"] = array + delta.astype("timedelta64[ms]")
+        dat = dat.set_index(["time"])
+
+    elif metadata["instmeta"]["VECBurstInterval"] == "CONTINUOUS":
+        start = dsvhd["time"][0].values
+        length = len(dat.Burst)
```

Comment:
Check for skipped samples

Reply:
Revamped this a bit so the code uses ensemble (sample) number when creating time. This way, it can account for any skipped samples.